tests/locators/hw22.spec.js

```diff
@@ -0,0 +1,543 @@
+            await signUpBtn.click()
+            // Как эти три строки можно вынести в beforeEach? Почему ругается на /
+
+            const form = page.locator('app-signup-modal form')
+            const nameInput = form.locator('#signupName')
+
+            await nameInput.fill("Aaaaaaaaaaaaaaaaaaaa")
+            await nameInput.blur()
+
+            const registerBtn = page.locator(".modal-footer > .btn")
+            await expect(registerBtn).toBeDisabled()
+        })
+
+        test("Name field > Empty field", async ({page}) => {
+            // Открыть форму регистрации нового пользователя
+            await page.goto("/")
+            const signUpBtn = page.locator(".hero-descriptor_btn")
+            await signUpBtn.click()
+            // Как эти три строки можно вынести в beforeEach? Почему ругается на /
+
+            const form = page.locator('app-signup-modal form')
+            const nameInput = form.locator('#signupName')
+
+            await nameInput.focus()
+            await nameInput.blur()
+
+            const inputNameTip = page.locator('.invalid-feedback > p')
+            await expect(inputNameTip).toContainText('Name required')
+            await expect(nameInput).toHaveCSS('border-color', 'rgb(220, 53, 69)')
+
+            const registerBtn = page.locator(".modal-footer > .btn")
+            await expect(registerBtn).toBeDisabled()
+        })
+
+        test("Name field > Name is invalid: Numbers", async ({page}) => {
+            // Открыть форму регистрации нового пользователя
+            await page.goto("/")
+            const signUpBtn = page.locator(".hero-descriptor_btn")
+            await signUpBtn.click()
+            // Как эти три строки можно вынести в beforeEach? Почему ругается на /
+
+            const form = page.locator('app-signup-modal form')
+            const nameInput = form.locator('#signupName')
+
+            await nameInput.fill("111")
+            await nameInput.blur()
+
+            const inputNameTip1 = page.locator('.invalid-feedback > :nth-child(1)')
+            await expect(inputNameTip1).toContainText('Name is invalid')
+            await expect(nameInput).toHaveCSS('border-color', 'rgb(220, 53, 69)')
+
+            const registerBtn = page.locator(".modal-footer > .btn")
+            await expect(registerBtn).toBeDisabled()
+        })
+
+        test("Name field > Name is invalid: Space", async ({page}) => {
+            // Открыть форму регистрации нового пользователя
+            await page.goto("/")
+            const signUpBtn = page.locator(".hero-descriptor_btn")
+            await signUpBtn.click()
+            // Как эти три строки можно вынести в beforeEach? Почему ругается на /
+
+            const form = page.locator('app-signup-modal form')
+            const nameInput = form.locator('#signupName')
+
+            await nameInput.fill("Anna Maria")
+            await nameInput.blur()
+
+            const inputNameTip1 = page.locator('.invalid-feedback > :nth-child(1)')
```

Comment:
use more precise selectors

for example
```
#signupName + .invalid-feedback
```

Reply:
fixed